models/Guard.cfc

```diff
@@ -0,0 +1,171 @@
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    function denies( required any permissions, struct additionalArgs = {} ) {
+        var context = preflight();
+
+        for ( var permission in arrayWrap( arguments.permissions ) ) {
+            var hasPermission = invoke(
+                context.user,
+                context.props.methodNames[ "hasPermission" ],
+                {
+                    "permission": permission,
+                    "additionalArgs": arguments.additionalArgs
+                }
+            );
+
+            if ( !hasPermission ) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public boolean function all( required any permissions, struct additionalArgs = {} ) {
+        var context = preflight();
+
+        for ( var permission in arrayWrap( arguments.permissions ) ) {
+            var hasPermission = invoke(
+                context.user,
+                context.props.methodNames[ "hasPermission" ],
+                {
+                    "permission": permission,
+                    "additionalArgs": arguments.additionalArgs
+                }
+            );
+
+            if ( !hasPermission ) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    public boolean function none( required any permissions, struct additionalArgs = {} ) {
+        var context = preflight();
+
+        for ( var permission in arrayWrap( arguments.permissions ) ) {
+            var hasPermission = invoke(
+                context.user,
+                context.props.methodNames[ "hasPermission" ],
+                {
+                    "permission": permission,
+                    "additionalArgs": arguments.additionalArgs
+                }
+            );
+
+            if ( hasPermission ) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    public void function authorize(
+        required any permissions,
+        struct additionalArgs = {},
+        string errorMessage
+    ) {
+        var context = preflight();
+
+        var failedPermission = "";
+        for ( var permission in arrayWrap( arguments.permissions ) ) {
+            var hasPermission = invoke(
+                context.user,
+                context.props.methodNames[ "hasPermission" ],
+                {
+                    "permission": permission,
+                    "additionalArgs": arguments.additionalArgs
+                }
+            );
+
+            if ( !hasPermission ) {
+                failedPermission = permission;
+                break;
+            }
+        }
+
+        if ( failedPermission != "" ) {
+            param arguments.errorMessage = "The logged in user is not authorized to access this resource";
+
+            var args = {
+                "failedPermission": failedPermission,
+                "user": context.user,
```

Comment:
This causes a stackoverflow without a getMemento() on it.
I know we can't assume everyone has a getMemento() but we need some way to make this serializable.

Reply:
We can either remove it here or specify that as part of the contract the user object needs a `getMemento`